File: backend/app/schemas/patient.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class PatientBase(BaseModel):
    id: int
    patient_code: str
    display_name: str
    is_synthetic: bool = True

    model_config = ConfigDict(from_attributes=True)


class PatientSummary(PatientBase):
    current_severity: str
    open_event_count: int
    latest_lab_observed_at: datetime | None
    risk_reasons: list[str] = Field(default_factory=list)
    oldest_unresolved_event_at: datetime | None = None
    driver_event_ids: list[int] = Field(default_factory=list)


class PatientDetail(PatientSummary):
    created_at: datetime


class PatientListResponse(BaseModel):
    items: list[PatientSummary]
    total: int

File: frontend/src/types/patient.ts
export type Severity = "NORMAL" | "REVIEW_REQUIRED" | "HIGH_RISK";

export interface PatientSummary {
  id: number;
  patient_code: string;
  display_name: string;
  is_synthetic: boolean;
  current_severity: Severity;
  open_event_count: number;
  latest_lab_observed_at: string | null;
  risk_reasons: string[];
  oldest_unresolved_event_at: string | null;
  driver_event_ids: number[];
}

export interface PatientDetail extends PatientSummary {
  created_at: string;
}

export interface PatientListResponse {
  items: PatientSummary[];
  total: number;
}
